code/modules/power/supermatter/supermatter.dm

```diff
@@ -678,6 +678,25 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 		if(produces_gas)
 			env.merge(removed)
 			air_update_turf()
+	if(takes_damage)
+		for(var/t in RANGE_TURFS(1, loc))
+			if(!isspaceturf(t))
+				continue
+			var/turf/turf_to_check = t
+			var/is_open =  TRUE
+			if(!LAZYLEN(turf_to_check.atmos_adjacent_turfs))
+				is_open = FALSE
+			if(is_open)
+				var/integrity = get_integrity()
+				if(integrity < 10)
+					damage += max((power * 0.0005) * DAMAGE_INCREASE_MULTIPLIER, 0)
+				else if(integrity < 25)
+					damage += max((power * 0.0009) * DAMAGE_INCREASE_MULTIPLIER, 0)
+				else if(integrity < 45)
+					damage += max((power * 0.005) * DAMAGE_INCREASE_MULTIPLIER, 0)
+				else if(integrity < 75)
+					damage += max((power * 0.002) * DAMAGE_INCREASE_MULTIPLIER, 0)
+			break
```

Comment:
You'll want to indent this in further, as it currently stands it will break after the first turf is processed, which also means ya should recheck your numbers, as there's a good chance they're wrong.

Reply:
Oops i did have this indented, but it seems that i reverted it before committing, anyway the numbers i did are for one exposed turf to space, i did try without the break and it was too much for the SM as it was delaminating in seconds, the numbers are correct.